tests/loader/test_astronet_loader.py

```diff
@@ -0,0 +1,183 @@
+"""Unit tests for the AstroNetLoader class.
+
+Authors: Travis Driver
+"""
+
+import unittest
+from pathlib import Path
+
+import numpy as np
+from gtsam import Pose3, PinholeCameraCal3Bundler
+
+from gtsfm.common.image import Image
+from gtsfm.loader.astronet_loader import AstroNetLoader
+from thirdparty.colmap.scripts.python.read_write_model import read_model
+import gtsfm.utils.io as io_utils
+
+TEST_DATA_ROOT = Path(__file__).resolve().parent.parent / "data"
+
+# These lists specify a subset of the Point3D IDs, and their corresponding SfmTrack index in the loader, for testing.
+# Specifically, given some index `j`, the data in the Point3D with ID `TEST_POINT3D_IDS[j]` should match that in the
+# SfmTrack indexed by `TEST_SFMTRACKS_INDICES[j]`.
+TEST_POINT3D_IDS = [7, 2560, 3063, 3252, 3424, 3534, 3653, 3786, 3920, 4062, 4420, 4698, 3311, 3963, 325]
+TEST_SFMTRACKS_INDICES = [0, 100, 200, 300, 400, 500, 600, 700, 800, 900, 1000, 1100, 1200, 1300, 1400]
+
+
+class TestAstroNetLoader(unittest.TestCase):
+    """Class containing tests for the AstroNetLoader."""
+
+    def setUp(self):
+        """Set up the loader for the test."""
+        super().setUp()
+
+        data_dir = TEST_DATA_ROOT / "astronet" / "test_2011212_opnav_022"
+        # gt_scene_mesh_path = data_dir / "vesta_5002.ply"
+
+        # Read in COLMAP-formatted data for comparison.
+        self.cameras, self.images, self.points3d = read_model(data_dir)
+
+        # Initialize Loader.
+        self.loader = AstroNetLoader(
+            data_dir,
+            use_gt_extrinsics=True,
+            use_gt_sfmtracks=True,
+            max_frame_lookahead=2,
+        )
+
+    def test_constructor_set_properties(self) -> None:
+        """Ensure that constructor sets class properties correctly."""
+        # assert self.loader._gt_scene_trimesh is not None
+        # assert self.loader._gt_scene_trimesh.vertices;shape[0] == 5002
```

Comment:
remove if unnecessary?

Reply:
I will be uncommenting this (and some other tests) in the next PR when I add trimesh. It would be useful to be able to keep them here so I don't have to rewrite them.